Gems/Atom/RPI/Code/Source/RPI.Public/ViewportContext.cpp

```diff
@@ -111,26 +122,38 @@ namespace AZ
                 {
                     SceneNotificationBus::Handler::BusConnect(m_rootScene->GetId());
                 }
-                m_currentPipeline.reset();
+                ResetCurrentPipeline();
                 UpdatePipelineView();
+                UpdatePipelineRefreshRate();
             }
 
             m_sceneChangedEvent.Signal(scene);
         }
 
-        void ViewportContext::RenderTick()
+        float ViewportContext::GetFpsLimit() const
         {
-            // add the current pipeline to next render tick if it's not already added.
-            if (m_currentPipeline && m_currentPipeline->GetRenderMode() != RenderPipeline::RenderMode::RenderOnce)
-            {
-                m_currentPipeline->AddToRenderTickOnce();
-            }
+            return m_fpsLimit;
         }
 
-        void ViewportContext::OnBeginPrepareRender()
+        void ViewportContext::SetFpsLimit(float fpsLimit)
         {
-            ViewportContextNotificationBus::Event(GetName(), &ViewportContextNotificationBus::Events::OnRenderTick);
-            ViewportContextIdNotificationBus::Event(GetId(), &ViewportContextIdNotificationBus::Events::OnRenderTick);
+            m_fpsLimit = fpsLimit;
+            UpdatePipelineRefreshRate();
+        }
+
+        float ViewportContext::GetTargetFramerate() const
```

Comment:
uber nit: Can ignore if this was like this before but we might want to uppercase Rate in FrameRate as they are two separate words (much like we have RefreshRate below) - so this becomes `GetTargetFrameRate` (https://en.wikipedia.org/wiki/Frame_rate)